pkg/splunk/controller/util.go

```diff
@@ -189,6 +189,32 @@ func MergePodSpecUpdates(current *corev1.PodSpec, revised *corev1.PodSpec, name
 	return result
 }
 
+// SortSlicesStatefulSet sorts required slices in a statefulSet
+func SortSlicesStatefulSet(current *corev1.PodSpec, name string) error {
+	scopedLog := log.WithName("SortSlicesStatefulSet").WithValues("name", name)
+
+	// Sort tolerations
+	splcommon.SortSlice(current.Tolerations, splcommon.SortFieldKey)
+
+	// Sort volumes
+	splcommon.SortSlice(current.Volumes, splcommon.SortFieldName)
+
+	// Sort slices inside container specs
+	for idx := range current.Containers {
+		// Sort container ports
+		splcommon.SortSlice(current.Containers[idx].Ports, splcommon.SortFieldContainerPort)
+
+		// Sort VolumeMounts
+		splcommon.SortSlice(current.Containers[idx].VolumeMounts, splcommon.SortFieldName)
+
+		// Sort env variables
+		splcommon.SortSlice(current.Containers[idx].Env, splcommon.SortFieldName)
+	}
+	scopedLog.Info("Successfully sorted slices in statefulSet")
+
+	return nil
```

Comment:
In the above comment, if SortSlice is returning a value, then needs to be populated here with a corresponding error?

Reply:
Same comment as above. Didn't find the need to return a value.